src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.h

```diff
@@ -0,0 +1,210 @@
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <string>
+#include <vector>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WebRTCTransportProvider {
+
+using ICEServerStruct     = Structs::ICEServerStruct::DecodableType;
+using WebRTCSessionStruct = Structs::WebRTCSessionStruct::Type;
+
+/**
+ * @brief
+ *   A delegate interface for the WebRTC Transport Provider cluster, allowing
+ *   OEM or application-specific logic to be implemented outside of the cluster
+ *   server itself. The server calls these methods to do the "real work."
+ */
+class Delegate
+{
+public:
+    virtual ~Delegate() = default;
+
+    struct SolicitOfferRequestArgs
+    {
+        uint16_t id;
+        StreamUsageEnum streamUsage;
+        Optional<DataModel::Nullable<uint16_t>> videoStreamId;
+        Optional<DataModel::Nullable<uint16_t>> audioStreamId;
+        Optional<DataModel::DecodableList<ICEServerStruct>> iceServers;
+        Optional<chip::CharSpan> iceTransportPolicy;
+        Optional<chip::BitMask<WebRTCMetadataOptionsBitmap>> metadataOptions;
+        NodeId peerNodeId;
+        FabricIndex peerFabricIndex;
+    };
+
+    struct ProvideOfferRequestArgs
+    {
+        uint16_t id;
+        StreamUsageEnum streamUsage;
+        chip::CharSpan sdp;
+        Optional<DataModel::Nullable<uint16_t>> videoStreamId;
+        Optional<DataModel::Nullable<uint16_t>> audioStreamId;
+        Optional<DataModel::DecodableList<ICEServerStruct>> iceServers;
+        Optional<chip::CharSpan> iceTransportPolicy;
+        Optional<chip::BitMask<WebRTCMetadataOptionsBitmap>> metadataOptions;
+        NodeId peerNodeId;
+        FabricIndex peerFabricIndex;
+    };
+
+    /**
+     * @brief
+     *     This method is called when the server receives the SolicitOffer command.
+     *
+     *     It is used by the requestor to ask the provider to initiate a new Offer/Answer
+     *     negotiation. The specification states that the provider may be in a standby mode
+     *     (low power), thus a "deferred offer" scenario might occur if the device needs time
+     *     to fully power resources needed for streaming.
+     *
+     * @param[in]  args             Structure containing all input arguments for the command.
+     * @param[out] outSession       On success, must be populated with the new session's details.
+     * @param[out] outDeferredOffer True if the device is in standby or needs to defer generating an Offer.
```

Comment:
Can this happen on any error result?
Or could we carve out a specific CHIP_ERROR to represent this being true?

Reading the documentation I can see no relation between return value and this parameter. If they are unrelated please update the documentation (i.e. can we really have all 4 combinations of `no_error+true`, `no_error + false`, `error + true`, `error + false`.

Reply:
Update the comments and make it clear that outDeferredOffer only has meaning on success